js/editTask.js

```diff
@@ -0,0 +1,53 @@
+import {
+  EDIT,
+  COMPLETE,
+  DELETE_TODO,
+  EDIT_SAVE,
+  EDIT_CHECK,
+  EDIT_DELETE,
+  DONE_ICON,
+  DELETE_ICON,
+} from "/js/constants.js";
+import handleSaveButtonClick from "/js/utility/SaveButtonAction.js";
+import handleDeleteButtonClick from "/js/utility/DeleteButtonAction.js";
+import handleCheckButtonClick from "/js/utility/CheckButtonAction.js";
+import createButton from "/js/utility/createButton.js";
+import toggleButtonDisplayByAttributes from "/js/utility/toggleButtonDisplayByAttributes.js";
+
+export default function editTask(todoItemElement, taskList, elementId) {
+  const attr = "data-job";
+
+  toggleButtonDisplayByAttributes(
+    todoItemElement,
+    `[${attr}="${DELETE_TODO}"]`,
+    `[${attr}="${EDIT}"]`,
+    `[${attr}="${COMPLETE}"]`
+  );
+
+  const textElement = todoItemElement.querySelector(".text");
+  textElement.contentEditable = true;
+  textElement.focus();
+
+  const saveButton = document.createElement("button");
+  saveButton.innerText = "Save";
+  saveButton.classList.add("saveButton");
+  saveButton.setAttribute("data-job", `${EDIT_SAVE}`);
+  saveButton.setAttribute("style", "display: inline-block");
```

Comment:
creating the save button at this stage (while editing) is not a good idea because it's not the responsibility of the editTask function. the DOM nodes for adding, editing, deleting a task should be created in the adding phase with proper state i.e some elements hidden, event handlers registered etc.